client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
+		const noticeId = `comment-notice-${ commentId }`;
+
+		dispatch( removeNotice( noticeId ) );
+
+		dispatch( successNotice( translate( 'Comment approved.' ), {
+			duration: 5000,
+			id: noticeId,
+			isPersistent: true,
+			button: translate( 'Undo' ),
+			onClick: () => dispatch( changeCommentStatus( siteId, postId, commentId, 'unapproved' ) )
```

Comment:
To go a bit further on the previous comment, I **do** think it would be ok if the comment becomes unliked. I also don't think "liked + unapproved" makes sense. Sorry if this is contrary to what I said. This is getting pretty confusing :)